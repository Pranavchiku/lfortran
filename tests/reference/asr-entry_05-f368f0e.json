{
    "basename": "asr-entry_05-f368f0e",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/entry_05.f90",
    "infile_hash": "6ba013faf5b219ebe9941b538cdb0e7566fcd7389ab49f3e09971eeb",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-entry_05-f368f0e.stdout",
    "stdout_hash": "5897174241e6f80972c9cd0fc9e1216ae88af353289c63bd070de440",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}