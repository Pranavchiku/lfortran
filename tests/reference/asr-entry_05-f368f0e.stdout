(TranslationUnit
    (SymbolTable
        1
        {
            dstzr:
                (Function
                    (SymbolTable
                        89
                        {
                            abstol:
                                (Variable
                                    89
                                    abstol
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            ftol:
                                (Function
                                    (SymbolTable
                                        90
                                        {
                                            ftol_return_var_name:
                                                (Variable
                                                    90
                                                    ftol_return_var_name
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            max:
                                                (ExternalSymbol
                                                    90
                                                    max
                                                    7 max
                                                    lfortran_intrinsic_math2
                                                    []
                                                    max
                                                    Private
                                                ),
                                            max@dmax:
                                                (ExternalSymbol
                                                    90
                                                    max@dmax
                                                    7 dmax
                                                    lfortran_intrinsic_math2
                                                    []
                                                    dmax
                                                    Private
                                                ),
                                            zx:
                                                (Variable
                                                    90
                                                    zx
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    ftol
                                    (FunctionType
                                        [(Real 8)]
                                        (Real 8)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [max@dmax]
                                    [(Var 90 zx)]
                                    [(=
                                        (Var 90 ftol_return_var_name)
                                        (RealBinOp
                                            (RealConstant
                                                0.500000
                                                (Real 8)
                                            )
                                            Mul
                                            (FunctionCall
                                                90 max@dmax
                                                90 max
                                                [((Var 89 abstol))
                                                ((RealBinOp
                                                    (Var 89 reltol)
                                                    Mul
                                                    (IntrinsicScalarFunction
                                                        Abs
                                                        [(Var 90 zx)]
                                                        0
                                                        (Real 8)
                                                        ()
                                                    )
                                                    (Real 8)
                                                    ()
                                                ))]
                                                (Real 8)
                                                ()
                                                ()
                                            )
                                            (Real 8)
                                            ()
                                        )
                                        ()
                                    )]
                                    (Var 90 ftol_return_var_name)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            n:
                                (Variable
                                    89
                                    n
                                    []
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            r:
                                (Variable
                                    89
                                    r
                                    [n]
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Real 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (Var 89 n))]
                                        PointerToDataArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            reltol:
                                (Variable
                                    89
                                    reltol
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            zx:
                                (Variable
                                    89
                                    zx
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    dstzr
                    (FunctionType
                        []
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(Return)]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                ),
            dzror:
                (Function
                    (SymbolTable
                        2
                        {
                            abstol:
                                (Variable
                                    2
                                    abstol
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            ftol:
                                (Function
                                    (SymbolTable
                                        5
                                        {
                                            ftol_return_var_name:
                                                (Variable
                                                    5
                                                    ftol_return_var_name
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            max:
                                                (ExternalSymbol
                                                    5
                                                    max
                                                    7 max
                                                    lfortran_intrinsic_math2
                                                    []
                                                    max
                                                    Private
                                                ),
                                            max@dmax:
                                                (ExternalSymbol
                                                    5
                                                    max@dmax
                                                    7 dmax
                                                    lfortran_intrinsic_math2
                                                    []
                                                    dmax
                                                    Private
                                                ),
                                            zx:
                                                (Variable
                                                    5
                                                    zx
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    ftol
                                    (FunctionType
                                        [(Real 8)]
                                        (Real 8)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [max@dmax]
                                    [(Var 5 zx)]
                                    [(=
                                        (Var 5 ftol_return_var_name)
                                        (RealBinOp
                                            (RealConstant
                                                0.500000
                                                (Real 8)
                                            )
                                            Mul
                                            (FunctionCall
                                                5 max@dmax
                                                5 max
                                                [((Var 2 abstol))
                                                ((RealBinOp
                                                    (Var 2 reltol)
                                                    Mul
                                                    (IntrinsicScalarFunction
                                                        Abs
                                                        [(Var 5 zx)]
                                                        0
                                                        (Real 8)
                                                        ()
                                                    )
                                                    (Real 8)
                                                    ()
                                                ))]
                                                (Real 8)
                                                ()
                                                ()
                                            )
                                            (Real 8)
                                            ()
                                        )
                                        ()
                                    )]
                                    (Var 5 ftol_return_var_name)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            n:
                                (Variable
                                    2
                                    n
                                    []
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            r:
                                (Variable
                                    2
                                    r
                                    [n]
                                    Unspecified
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Real 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (Var 2 n))]
                                        PointerToDataArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            reltol:
                                (Variable
                                    2
                                    reltol
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            zx:
                                (Variable
                                    2
                                    zx
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Real 8)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    dzror
                    (FunctionType
                        [(Integer 4)
                        (Array
                            (Real 4)
                            [((IntegerConstant 1 (Integer 4))
                            (FunctionParam
                                0
                                (Integer 4)
                                ()
                            ))]
                            PointerToDataArray
                        )]
                        ()
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    [(Var 2 n)
                    (Var 2 r)]
                    [(=
                        (Var 2 r)
                        (RealUnaryMinus
                            (RealConstant
                                83.030000
                                (Real 4)
                            )
                            (Real 4)
                            (RealConstant
                                -83.030000
                                (Real 4)
                            )
                        )
                        ()
                    )
                    (Return)]
                    ()
                    Public
                    .false.
                    .false.
                    ()
                ),
            dzror_saved_variables:
                (Module
                    (SymbolTable
                        3
                        {
                            
                        })
                    dzror_saved_variables
                    []
                    .false.
                    .false.
                ),
            entry_05:
                (Program
                    (SymbolTable
                        4
                        {
                            i:
                                (Variable
                                    4
                                    i
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            r:
                                (Variable
                                    4
                                    r
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Array
                                        (Real 4)
                                        [((IntegerConstant 1 (Integer 4))
                                        (IntegerConstant 10 (Integer 4)))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    entry_05
                    []
                    [(SubroutineCall
                        1 dzror
                        ()
                        [((IntegerConstant 10 (Integer 4)))
                        ((ArrayPhysicalCast
                            (Var 4 r)
                            FixedSizeArray
                            PointerToDataArray
                            (Array
                                (Real 4)
                                [((IntegerConstant 1 (Integer 4))
                                (IntegerConstant 10 (Integer 4)))]
                                PointerToDataArray
                            )
                            ()
                        ))]
                        ()
                    )
                    (DoLoop
                        ()
                        ((Var 4 i)
                        (IntegerConstant 1 (Integer 4))
                        (IntegerConstant 10 (Integer 4))
                        ())
                        [(If
                            (RealCompare
                                (IntrinsicScalarFunction
                                    Abs
                                    [(RealBinOp
                                        (ArrayItem
                                            (Var 4 r)
                                            [(()
                                            (Var 4 i)
                                            ())]
                                            (Real 4)
                                            ColMajor
                                            ()
                                        )
                                        Sub
                                        (RealUnaryMinus
                                            (RealConstant
                                                83.030000
                                                (Real 4)
                                            )
                                            (Real 4)
                                            (RealConstant
                                                -83.030000
                                                (Real 4)
                                            )
                                        )
                                        (Real 4)
                                        ()
                                    )]
                                    0
                                    (Real 4)
                                    ()
                                )
                                Gt
                                (RealConstant
                                    0.000000
                                    (Real 4)
                                )
                                (Logical 4)
                                ()
                            )
                            [(ErrorStop
                                ()
                            )]
                            []
                        )]
                    )
                    (Print
                        [(Var 4 r)]
                        ()
                        ()
                    )]
                ),
            iso_fortran_env:
                (IntrinsicModule lfortran_intrinsic_iso_fortran_env),
            lfortran_intrinsic_builtin:
                (IntrinsicModule lfortran_intrinsic_builtin),
            lfortran_intrinsic_math2:
                (IntrinsicModule lfortran_intrinsic_math2),
            lfortran_intrinsic_math3:
                (IntrinsicModule lfortran_intrinsic_math3)
        })
    []
)
